hail/python/hail/expr/aggregators/aggregators.py

```diff
@@ -235,7 +235,7 @@ def approx_cdf(expr, k=100):
     :class:`.StructExpression`
         Struct containing `values` and `ranks` arrays.
     """
-    return _agg_func('ApproxCDF', [expr], tstruct(values=tarray(expr.dtype), ranks=tarray(tint64)), constructor_args=[k])
+    return _agg_func('ApproxCDF', [expr], tstruct(values=tarray(expr.dtype), ranks=tarray(tint64), _compaction_counts=tarray(tint32)), constructor_args=[k])
```

Comment:
I don't super like that we've got undocumented fields in the return type; would it be reasonable to document?

Reply:
What we really want is for the `ApproxCDF` aggregator to return a custom hail type, that hides implementation details like compaction counts, and supports methods like estimating quantiles or getting error bounds. But that should probably wait until physical types are more developed. Or maybe it's reasonable to do now; I haven't looked into it.

I can add documentation that just says this is an implementation detail, so users aren't confused by it.